test/models/deploy_spec_test.rb

```diff
@@ -684,10 +684,104 @@ class DeploySpecTest < ActiveSupport::TestCase
+      package_json = Pathname.new('/tmp/fake_package.json')
+      package_json.write('{"publishConfig": "foo"}')
+
+      @spec.expects(:package_json).returns(package_json)
+      assert_equal "foo", @spec.publish_config
+    end
+
+    test '#valid_publish_config_access? is false when publishConfig.access is invalid' do
+      @spec.stubs(:publish_config_access).returns('foo')
+      refute @spec.valid_publish_config_access?
+    end
+
+    test '#valid_publish_config_access? is true when publishConfig.access is public or restricted' do
+      @spec.stubs(:publish_config_access).returns('public')
+      assert @spec.valid_publish_config_access?
+
+      @spec.stubs(:publish_config_access).returns('restricted')
+      assert @spec.valid_publish_config_access?
+    end
+
+    test '#publish_config_access is restricted when publishConfig.access is missing' do
+      package_json = Pathname.new('/tmp/fake_package.json')
+      package_json.write('{"publishConfig": {}}')
+
+      @spec.expects(:package_json).returns(package_json)
+      assert_equal 'restricted', @spec.publish_config_access
+    end
+
+    test '#publish_config_access returns publishConfig.access in package.json' do
+      package_json = Pathname.new('/tmp/fake_package.json')
+      package_json.write('{"publishConfig": {
+          "access": "foo"
+        }
+      }')
+      @spec.expects(:package_json).returns(package_json)
+      assert_equal "foo", @spec.publish_config_access
+    end
+
+    test "#package_scoped_when_private? is false when private packages are not scoped" do
+      @spec.stubs(:scoped_package?).returns(false)
+      @spec.stubs(:publish_config_access).returns("restricted")
+      refute @spec.package_scoped_when_private?
+    end
+
+    test "#package_scoped_when_private? is true when private packages are scoped" do
+      @spec.stubs(:scoped_package?).returns(true)
+      @spec.stubs(:publish_config_access).returns("restricted")
+      assert @spec.package_scoped_when_private?
+    end
+
+    test "#package_scoped_when_private? is true when public packages are not scoped" do
+      @spec.stubs(:scoped_package?).returns(false)
+      @spec.stubs(:publish_config_access).returns("public")
+      assert @spec.package_scoped_when_private?
+    end
+
     test '#publish_npm_package checks if version tag exists, and then invokes npm deploy script' do
       @spec.stubs(:npm?).returns(true)
       @spec.stubs(:package_version).returns('1.0.0')
-      assert_equal ['assert-npm-version-tag', 'npm publish --tag latest'], @spec.deploy_steps
+      @spec.stubs(:valid_publish_config?).returns(true)
+      @spec.stubs(:publish_config_access).returns('restricted')
+      @spec.stubs(:generate_local_npmrc).returns(true)
+      assert_equal ['assert-npm-version-tag', 'npm publish --tag latest --access restricted'], @spec.deploy_steps
+    end
+
+    test '#npmrc_contents returns a private package configuration' do
```

Comment:
Under what circumstances?

Reply:
`#npmrc_contents returns a scoped private package configuration when the package is scoped and private`

PR updated